isis/src/control/objs/ControlNet/ControlNet.cpp

```diff
@@ -1844,8 +1842,23 @@ namespace Isis {
       i2.next().value()->parentNetwork = &other;
     }
 
-    emit networkModified(ControlNet::Swapped);
+    m_vertexMap.clear();
+    VertexIterator v, vend;
+    for (boost::tie(v, vend) = vertices(m_controlGraph); v != vend; ++v) {
+      ImageVertex imVertex = *v;
+      QString serialNum = m_controlGraph[*v].serial;
+      m_vertexMap[serialNum] = imVertex;
+    }
+
+    other.m_vertexMap.clear();
+    VertexIterator v2, vend2;
+    for (boost::tie(v2, vend2) = vertices(other.m_controlGraph); v2 != vend2; ++v2) {
+      ImageVertex imVertex = *v2;
+      QString serialNum = other.m_controlGraph[*v2].serial;
+      other.m_vertexMap[serialNum] = imVertex;
+    }
 
+    emit networkModified(ControlNet::Swapped);
```

Comment:
We also need the other network to emit this signal, I'm not sure what the best way to do that is though? For right now, I don't think this is being used anywhere that it matters, but it may come up in the future.